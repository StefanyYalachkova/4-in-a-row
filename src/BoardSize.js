import React from "react";
import { Button, TextField } from '@mui/material';

const BoardSize = (props) => {

    const handleConfirm = (event) => {
        event.preventDefault();
        props.onClick();
    };

    return (
        <div>
            <form>
                <h1>Set board size :</h1>
                <span id="rowsField">Rows:</span>
                <TextField
                    type="number"
                    name="rows"
                    onChange={props.setSize}
                    helperText={props.isRowNumberValid ? '' : <span id="errorMessage">Enter a number greater than 4.</span>}
                />
                <span id="colsField">Columns:</span>
                <TextField
                    type="number"
                    name="columns"
                    onChange={props.setSize}
                    helperText={props.isColNumberValid ? '' : <span id="errorMessage">Enter a number greater than 4.</span>}
                />
                <span id="nInARowField">N in a row:</span>
                <TextField
                    type="number"
                    name="nInARows"
                    onChange={props.setSize}
                    helperText={props.isNinARowValid ? '' : <span id="errorMessage">Enter a number greater than 2.</span>}
                />
                <Button disabled={props.shouldDisabledButtonBoardSize} variant="contained" type="submit" onClick={handleConfirm}> Confirm </Button>
            </form>
        </div>
    );
};

export { BoardSize };