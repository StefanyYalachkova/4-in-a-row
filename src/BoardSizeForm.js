import React from "react";
import { Button, TextField } from '@mui/material';
import { MIN_VALID_ROW_COL_NUMBER, MIN_VALID_WINCOUNT_NUMBER } from "./constValue";

const BoardSizeForm = (props) => {
    const { setSize, isRowNumberValid, isColNumberValid, isWinCountValid, winCount, rowSize, colSize } = props;
    const shouldDisableBoardSizeButton = rowSize < MIN_VALID_ROW_COL_NUMBER || colSize < MIN_VALID_ROW_COL_NUMBER || winCount < MIN_VALID_WINCOUNT_NUMBER;

    const handleConfirm = (event) => {
        event.preventDefault();
        props.onClick();
    };

    return (
        <form>
            <h1>Set board size :</h1>
            <span id="rowsField">Rows:</span>
            <TextField
                type="number"
                name="rows"
                onChange={setSize}
                helperText={isRowNumberValid ? '' : <span className="errorMessage">Enter a number greater than 4.</span>}
            />
            <span id="colsField">Columns:</span>
            <TextField
                type="number"
                name="columns"
                onChange={setSize}
                helperText={isColNumberValid ? '' : <span className="errorMessage">Enter a number greater than 4.</span>}
            />
            <span id="winCountField">N in a row:</span>
            <TextField
                type="number"
                name="winCount"
                onChange={setSize}
                helperText={isWinCountValid ? '' : <span className="errorMessage">Enter a number greater than 2.</span>}
            />
            <Button disabled={shouldDisableBoardSizeButton} variant="contained" type="submit" onClick={handleConfirm}> Confirm </Button>
        </form>
    );
};

export { BoardSizeForm };
