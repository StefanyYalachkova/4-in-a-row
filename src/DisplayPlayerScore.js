import React from "react";

const DisplayPlayerScore = (props) => {

    return (
        <div>
            <h2>{props.player.name}: {props.player.score}</h2>
        </div>
    );
};

export { DisplayPlayerScore };