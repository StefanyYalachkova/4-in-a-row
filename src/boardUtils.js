import { Grid } from "@mui/material";
import { SVGBallPlayer1 } from "./SVGBallPlayer1";
import { SVGBallPlayer2 } from "./SVGBallPlayer2";

const getIsBallObjectEmpty = (ballObject) => {
    return !(ballObject.isPlayerOneBall || ballObject.isPlayerTwoBall);
};

const areBallObjectsEqual = (ballObject1, ballObject2) => {
    return ballObject1.isPlayerOneBall === ballObject2.isPlayerOneBall
        && ballObject1.isPlayerTwoBall === ballObject2.isPlayerTwoBall;
};

const getBoardInitialState = (rowSize, colSize) => {
    const matrixOfObjects = [];

    for (let i = 0; i < rowSize; i++) {
        matrixOfObjects[i] = [];
        for (let j = 0; j < colSize; j++) {
            matrixOfObjects[i][j] = ({ isPlayerOneBall: false, isPlayerTwoBall: false });
        };
    };

    return matrixOfObjects;
};

const displayBoard = (boardState, keyPrefix) => {
    let result = [];
    const rowCount = boardState.length;
    const colCount = boardState[0].length;

    let getColElement = (keyPrefix, rowIndex, col) => {
        return (
            <Grid className={'cow-container'} item xs={1} key={`${keyPrefix}-${rowIndex}-${col}`}>
                <svg viewBox="0 0 24 24">
                    <rect width="30" height="30" style={{ fill: "rgb(0,0,255)", strokeWidth: 0, stroke: "rgb(0,0,0)" }} />
                    <DisplayBall element={boardState[rowIndex][col]} />
                </svg>
            </Grid>
        );
    };

    const getColumns = (keyPrefix, rowIndex) => {
        const columns = [];

        for (let col = 0; col < colCount; col++) {
            columns.push(getColElement(keyPrefix, rowIndex, col));
        };

        return columns;
    };

    const getRowElement = (key, rowIndex) => (
        <Grid className={'row-container'} xs={9} key={key}>
            {getColumns(keyPrefix, rowIndex)}
        </Grid>
    );

    for (let row = 0; row < rowCount; row++) {
        result.push(getRowElement(`${keyPrefix}-${row}`, row));
    };

    return result;
};

const DisplayBall = (props) => {
    const { element } = props;

    let result = (
        <svg>
            <circle
                cx="12" cy="12" r="8"
                strokeWidth="6" stroke="white"
                fill="white"
            />
        </svg>
    );

    if (element.isPlayerOneBall || element.isPlayerTwoBall) {
        result = (
            element.isPlayerOneBall
                ? <SVGBallPlayer1 />
                : <SVGBallPlayer2 />
        );
    };

    return result;
};

export { getIsBallObjectEmpty, areBallObjectsEqual, getBoardInitialState, displayBoard, DisplayBall };
