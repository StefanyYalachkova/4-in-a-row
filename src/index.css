body {
  padding: 20px;
  background-color: #f1d2e0e1;
}

button[type=submit] {
  font-family: sans-serif;
  background-color: rgb(0, 0, 255);
  padding: 10px;
  border-radius: 8px;
  height: 100%;
  cursor: pointer;
  margin: 5px;
  position: relative;
  left: 5%;
}

form,
h1 {
  text-align: center;
  padding: 15px;
}

h2 {
  text-align: inherit;
}

.playerOne-container {
  display: flex;
  align-items: center;
  justify-content: right;
}

.playerTwo-container {
  display: flex;
  align-items: center;
  justify-content: left;
}

.board-container {
  align-items: center;
  justify-content: center;
}

.row-container {
  display: flex;
  margin: -5px;
  align-items: center;
  justify-content: flex-end
}

.balls-container {
  display: flex;
  border: solid;
  align-items: center;
  justify-content: center;
}

.balls {
  width: auto;
  height: auto;

}

.firstLine-container {
  padding-bottom: 8px;
  align-items: center;
  justify-content: center;
  border-style: hidden;
  outline: none;
}

.winner-container {
  text-align: right;
  padding: 50px;
}

.errorMessage {
  color: red;
  font-size: 15px;
}

#playerField,
#rowsField,
#colsField,
#winCountField {
  padding: 20px;
  font-weight: bold;
  font-size: 20px;
}

.player-container {
  text-align: center;
  font-size: xx-large;
}
