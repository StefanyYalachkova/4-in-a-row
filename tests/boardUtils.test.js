import { areBallObjectsEqual, DisplayBall, getBoardInitialState, getIsBallObjectEmpty } from "../src/boardUtils";
import renderer from 'react-test-renderer';
import { render } from "@testing-library/react";
import { SVGBallPlayer1 } from "../src/SVGBallPlayer1";
import { SVGBallPlayer2 } from "../src/SVGBallPlayer2";

describe("getIsBallObjectEmpty", () => {
    it('checking is ball object empty', () => {
        expect(getIsBallObjectEmpty({ isPlayerOneBall: false, isPlayerTwoBall: false })).toBe(true);
        expect(getIsBallObjectEmpty({ isPlayerOneBall: true, isPlayerTwoBall: false })).toBe(false);
    });
});

describe("areBallObjectsEqual", () => {
    it('checking are ball objects equal', () => {
        expect(areBallObjectsEqual({ isPlayerOneBall: true, isPlayerTwoBall: false }, { isPlayerOneBall: true, isPlayerTwoBall: false })).toBe(true);
        expect(areBallObjectsEqual({ isPlayerOneBall: true, isPlayerTwoBall: false }, { isPlayerOneBall: false, isPlayerTwoBall: false })).toBe(false);
    });
});

describe("getBoardInitialState", () => {
    it('checking board initial state', () => {
        const expectedResult = [
            [{ isPlayerOneBall: false, isPlayerTwoBall: false }, { isPlayerOneBall: false, isPlayerTwoBall: false }]
        ];
        expect(getBoardInitialState(1, 2)).toEqual(expectedResult);
    });
    it('checking board with empty array', () => {
        const emptyArray = [];
        expect(getBoardInitialState(0, 0)).toEqual(emptyArray);
    });
});

describe('DisplayBall', () => {
    it('renders correctly PlayerOneBall', () => {
        const element = { isPlayerOneBall: true, isPlayerTwoBall: false };
        let result = (
            <svg>
                <circle
                    cx="12" cy="12" r="8"
                    strokeWidth="6" stroke="white"
                    fill="white"
                />
            </svg>
        );

        if (element.isPlayerOneBall || element.isPlayerTwoBall) {
            result = (
                element.isPlayerOneBall
                    ? <SVGBallPlayer1 />
                    : <SVGBallPlayer2 />
            );
        };

        const tree = renderer
            .create(<DisplayBall element={element} result={result} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders correctly isPlayerTwoBall', () => {
        const element = { isPlayerOneBall: false, isPlayerTwoBall: true };
        let result = (
            <svg>
                <circle
                    cx="12" cy="12" r="8"
                    strokeWidth="6" stroke="white"
                    fill="white"
                />
            </svg>
        );

        if (element.isPlayerOneBall || element.isPlayerTwoBall) {
            result = (
                element.isPlayerOneBall
                    ? <SVGBallPlayer1 />
                    : <SVGBallPlayer2 />
            );
        };

        const tree = renderer
            .create(<DisplayBall element={element} result={result} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders correctly empty circle', () => {
        const element = { isPlayerOneBall: false, isPlayerTwoBall: false };
        let result = (
            <svg>
                <circle
                    cx="12" cy="12" r="8"
                    strokeWidth="6" stroke="white"
                    fill="white"
                />
            </svg>
        );

        if (element.isPlayerOneBall || element.isPlayerTwoBall) {
            result = (
                element.isPlayerOneBall
                    ? <SVGBallPlayer1 />
                    : <SVGBallPlayer2 />
            );
        };

        const tree = renderer
            .create(<DisplayBall element={element} result={result} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('displayBoard test', () => {
    it('renders correctly displayBoard', () => {
        const boardState = [
            { isPlayerOneBall: false, isPlayerTwoBall: false },
            { isPlayerOneBall: false, isPlayerTwoBall: false },
            { isPlayerOneBall: false, isPlayerTwoBall: true }
        ];
        const keyPrefix = 'board';
        const displayBoard = jest.fn();

        render(displayBoard(boardState, keyPrefix))
        expect(displayBoard).toBeCalled();
    });
});
