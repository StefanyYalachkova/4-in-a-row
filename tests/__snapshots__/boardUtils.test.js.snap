// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayBall renders correctly PlayerOneBall 1`] = `
<svg
  className="balls"
  id="svg2495"
  version="1.0"
  viewBox="0 0 370 370"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs
    id="defs2497"
  >
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2641"
      x1="266.01"
      x2="346.31"
      y1="365.8"
      y2="357.98"
    >
      <stop
        id="stop3276"
        offset="0"
        stopColor="#f57936"
      />
      <stop
        id="stop3278"
        offset="1"
        stopColor="#f57936"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2643"
      x1="274.49"
      x2="282.3"
      y1="342.5"
      y2="305.03"
    >
      <stop
        id="stop3286"
        offset="0"
        stopColor="#f3813b"
      />
      <stop
        id="stop3288"
        offset="1"
        stopColor="#f3813b"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2645"
      x1="245.71"
      x2="250.36"
      y1="377.69"
      y2="321.84"
    >
      <stop
        id="stop3296"
        offset="0"
        stopColor="#f9a465"
      />
      <stop
        id="stop3298"
        offset="1"
        stopColor="#f9a465"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2647"
      x1="225.99"
      x2="165.66"
      y1="400.07"
      y2="383.22"
    >
      <stop
        id="stop3306"
        offset="0"
        stopColor="#faa668"
      />
      <stop
        id="stop3308"
        offset="1"
        stopColor="#faa668"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2649"
      x1="199.3"
      x2="161.36"
      y1="420.02"
      y2="447.11"
    >
      <stop
        id="stop3316"
        offset="0"
        stopColor="#f17e39"
      />
      <stop
        id="stop3318"
        offset="1"
        stopColor="#f17e39"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2651"
      x1="218.14"
      x2="213.35"
      y1="429.69"
      y2="481.32"
    >
      <stop
        id="stop3326"
        offset="0"
        stopColor="#f67a34"
      />
      <stop
        id="stop3328"
        offset="1"
        stopColor="#f67a34"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2653"
      x1="235.32"
      x2="277.47"
      y1="407.04"
      y2="450.98"
    >
      <stop
        id="stop3336"
        offset="0"
        stopColor="#fc9f59"
      />
      <stop
        id="stop3338"
        offset="1"
        stopColor="#fc9f59"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2655"
      x1="264.82"
      x2="314.27"
      y1="375.79"
      y2="407.69"
    >
      <stop
        id="stop3346"
        offset="0"
        stopColor="#fc9f59"
      />
      <stop
        id="stop3348"
        offset="1"
        stopColor="#fc9f59"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2657"
      x1="465.71"
      x2="318.1"
      y1="527.25"
      y2="408.09"
    >
      <stop
        id="stop3356"
        offset="0"
        stopColor="#9f410d"
      />
      <stop
        id="stop3358"
        offset="1"
        stopColor="#9f410d"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2659"
      x1="442.48"
      x2="334.06"
      y1="360.5"
      y2="358.8"
    >
      <stop
        id="stop3366"
        offset="0"
        stopColor="#a1420e"
      />
      <stop
        id="stop3368"
        offset="1"
        stopColor="#a1420e"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2661"
      x1="325.34"
      x2="302.92"
      y1="277.19"
      y2="309.85"
    >
      <stop
        id="stop3376"
        offset="0"
        stopColor="#a1420e"
      />
      <stop
        id="stop3378"
        offset="1"
        stopColor="#a1420e"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2663"
      x1="202.07"
      x2="221.25"
      y1="285.65"
      y2="314.59"
    >
      <stop
        id="stop3386"
        offset="0"
        stopColor="#a1420e"
      />
      <stop
        id="stop3388"
        offset="1"
        stopColor="#a1420e"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2665"
      x1="128.57"
      x2="167.59"
      y1="365.34"
      y2="383.22"
    >
      <stop
        id="stop3396"
        offset="0"
        stopColor="#a1420e"
      />
      <stop
        id="stop3398"
        offset="1"
        stopColor="#a1420e"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2667"
      x1="362.31"
      x2="265.11"
      y1="608.8"
      y2="446.23"
    >
      <stop
        id="stop3406"
        offset="0"
        stopColor="#a1420d"
      />
      <stop
        id="stop3408"
        offset="1"
        stopColor="#a1420d"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2669"
      x1="224.3"
      x2="216.96"
      y1="615.29"
      y2="488.47"
    >
      <stop
        id="stop3416"
        offset="0"
        stopColor="#a1420d"
      />
      <stop
        id="stop3418"
        offset="1"
        stopColor="#a1420d"
        stopOpacity="0"
      />
    </linearGradient>
    <linearGradient
      gradientTransform="translate(-112.52 -271)"
      gradientUnits="userSpaceOnUse"
      id="linearGradient2671"
      x1="118.66"
      x2="161.36"
      y1="489.24"
      y2="450.93"
    >
      <stop
        id="stop3426"
        offset="0"
        stopColor="#a3430e"
      />
      <stop
        id="stop3428"
        offset="1"
        stopColor="#a3430e"
        stopOpacity="0"
      />
    </linearGradient>
  </defs>
  <g
    id="layer1"
  >
    <g
      id="g2614"
      transform="matrix(-1 0 0 1 369.17 0)"
    >
      <path
        d="m482.08 468.87a170.62 170.62 0 1 1 -341.25 0 170.62 170.62 0 1 1 341.25 0z"
        id="path2474"
        transform="matrix(1.0459 0 0 1.0459 -141.15 -305.75)"
      />
      <path
        d="m129.41 119.31l-23.46 28.9c57.19 41.9 32.86 187.63 96.28 214 44.87-4.41 84.89-25.47 113.78-56.9-57.83-69.47-117.64-135.6-186.6-186z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2458"
      />
      <path
        d="m163.13 86.808l-25.28 25.282c83.76 71.01 136.76 130.49 182.35 187.19l0.75 0.43c26.26-31.07 42.09-71.24 42.09-115.09 0-3.55-0.11-7.07-0.31-10.56-31.72-64-150.87-43.02-199.6-87.252z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2460"
      />
      <path
        d="m268.66 37.745c-39.81 0.398-90.39 26.012-95.31 40.656 1.62 36.759 170.6 25.989 188.38 86.659l0.25 0.03c-5.59-51.2-32.84-95.91-72.41-124.75-6.46-1.878-13.55-2.668-20.91-2.595z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2462"
      />
      <path
        d="m184.6 6.1826c-8.95 0-17.77 0.6624-26.37 1.9375-26.55 28.735-12.38 54.549 1.87 62.438 12.17 8.663 43.55-45.581 122.22-35.219-28.08-18.426-61.65-29.156-97.72-29.156z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2464"
      />
      <path
        d="m103.26 158.68c-1.63 0.03-3.333 0.36-5.127 0.97-23.257 18.25-65.801 106.07-40.937 149.85 0.005 0-0.005 0.02 0 0.03 32.392 33.03 77.524 53.53 127.4 53.53 1.37 0 2.73 0 4.1-0.03-53.74-38.84-42.04-193.68-85.44-204.35z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2466"
      />
      <path
        d="m60.321 136.15c-19.536 0.62-42.335 10.86-53.063 33.13l-0.5622 1.34c-0.3594 4.62-0.5625 9.29-0.5625 14 0 44.15 16.068 84.55 42.657 115.72-24.609-77.25 50.266-145.17 42.125-152.72-5.382-7.8-17.227-11.89-30.594-11.47z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2468"
      />
      <path
        d="m54.665 62.37c-25.213 26.784-42.28 61.34-47.125 99.69 18.967-31.72 47.261-43.24 93-19.25l23.47-29.5c-15.33-13.829-47.642-37.702-69.345-50.94z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2470"
      />
      <path
        d="m150.51 9.4326c-34.96 6.7684-66.285 23.757-90.752 47.718l0.438 0.782c25.737 11.719 47.754 30.929 70.434 48.747l27.07-25.279c-17.64-17.784-31.24-37.202-7.19-71.968z"
        fill="#d65a1e"
        fillRule="evenodd"
        id="path2472"
      />
      <path
        d="m268.66 37.745c-39.81 0.398-90.39 26.012-95.31 40.656 1.62 36.759 170.6 25.989 188.38 86.659l0.25 0.03c-5.59-51.2-32.84-95.91-72.41-124.75-6.46-1.878-13.55-2.668-20.91-2.595z"
        fill="url(#linearGradient2641)"
        fillRule="evenodd"
        id="path3272"
      />
      <path
        d="m184.6 6.1826c-8.95 0-17.77 0.6624-26.37 1.9375-26.55 28.735-12.38 54.549 1.87 62.438 12.17 8.663 43.55-45.581 122.22-35.219-28.08-18.426-61.65-29.156-97.72-29.156z"
        fill="url(#linearGradient2643)"
        fillRule="evenodd"
        id="path3282"
      />
      <path
        d="m150.51 9.4326c-34.96 6.7684-66.285 23.757-90.752 47.718l0.438 0.782c25.737 11.719 47.754 30.929 70.434 48.747l27.07-25.279c-17.64-17.784-31.24-37.202-7.19-71.968z"
        fill="url(#linearGradient2645)"
        fillRule="evenodd"
        id="path3292"
      />
      <path
        d="m54.665 62.37c-25.213 26.784-42.28 61.34-47.125 99.69 18.967-31.72 47.261-43.24 93-19.25l23.47-29.5c-15.33-13.829-47.642-37.702-69.345-50.94z"
        fill="url(#linearGradient2647)"
        fillRule="evenodd"
        id="path3302"
      />
      <path
        d="m60.321 136.15c-19.536 0.62-42.335 10.86-53.063 33.13l-0.5622 1.34c-0.3594 4.62-0.5625 9.29-0.5625 14 0 44.15 16.068 84.55 42.657 115.72-24.609-77.25 50.266-145.17 42.125-152.72-5.382-7.8-17.227-11.89-30.594-11.47z"
        fill="url(#linearGradient2649)"
        fillRule="evenodd"
        id="path3312"
      />
      <path
        d="m103.26 158.68c-1.63 0.03-3.333 0.36-5.127 0.97-23.257 18.25-65.801 106.07-40.937 149.85 0.005 0-0.005 0.02 0 0.03 32.392 33.03 77.524 53.53 127.4 53.53 1.37 0 2.73 0 4.1-0.03-53.74-38.84-42.04-193.68-85.44-204.35z"
        fill="url(#linearGradient2651)"
        fillRule="evenodd"
        id="path3322"
      />
      <path
        d="m129.41 119.31l-23.46 28.9c57.19 41.9 32.86 187.63 96.28 214 44.87-4.41 84.89-25.47 113.78-56.9-57.83-69.47-117.64-135.6-186.6-186z"
        fill="url(#linearGradient2653)"
        fillRule="evenodd"
        id="path3332"
      />
      <path
        d="m163.13 86.808l-25.28 25.282c83.76 71.01 136.76 130.49 182.35 187.19l0.75 0.43c26.26-31.07 42.09-71.24 42.09-115.09 0-3.55-0.11-7.07-0.31-10.56-31.72-64-150.87-43.02-199.6-87.252z"
        fill="url(#linearGradient2655)"
        fillRule="evenodd"
        id="path3342"
      />
      <path
        d="m163.13 86.808l-25.28 25.282c83.76 71.01 136.76 130.49 182.35 187.19l0.75 0.43c26.26-31.07 42.09-71.24 42.09-115.09 0-3.55-0.11-7.07-0.31-10.56-31.72-64-150.87-43.02-199.6-87.252z"
        fill="url(#linearGradient2657)"
        fillRule="evenodd"
        id="path3352"
      />
      <path
        d="m268.66 37.745c-39.81 0.398-90.39 26.012-95.31 40.656 1.62 36.759 170.6 25.989 188.38 86.659l0.25 0.03c-5.59-51.2-32.84-95.91-72.41-124.75-6.46-1.878-13.55-2.668-20.91-2.595z"
        fill="url(#linearGradient2659)"
        fillRule="evenodd"
        id="path3362"
      />
      <path
        d="m184.6 6.1826c-8.95 0-17.77 0.6624-26.37 1.9375-26.55 28.735-12.38 54.549 1.87 62.438 12.17 8.663 43.55-45.581 122.22-35.219-28.08-18.426-61.65-29.156-97.72-29.156z"
        fill="url(#linearGradient2661)"
        fillRule="evenodd"
        id="path3372"
      />
      <path
        d="m150.51 9.4326c-34.96 6.7684-66.285 23.757-90.752 47.718l0.438 0.782c25.737 11.719 47.754 30.929 70.434 48.747l27.07-25.279c-17.64-17.784-31.24-37.202-7.19-71.968z"
        fill="url(#linearGradient2663)"
        fillRule="evenodd"
        id="path3382"
      />
      <path
        d="m54.665 62.37c-25.213 26.784-42.28 61.34-47.125 99.69 18.967-31.72 47.261-43.24 93-19.25l23.47-29.5c-15.33-13.829-47.642-37.702-69.345-50.94z"
        fill="url(#linearGradient2665)"
        fillRule="evenodd"
        id="path3392"
      />
      <path
        d="m129.41 119.31l-23.46 28.9c57.19 41.9 32.86 187.63 96.28 214 44.87-4.41 84.89-25.47 113.78-56.9-57.83-69.47-117.64-135.6-186.6-186z"
        fill="url(#linearGradient2667)"
        fillRule="evenodd"
        id="path3402"
      />
      <path
        d="m103.26 158.68c-1.63 0.03-3.333 0.36-5.127 0.97-23.257 18.25-65.801 106.07-40.937 149.85 0.005 0-0.005 0.02 0 0.03 32.392 33.03 77.524 53.53 127.4 53.53 1.37 0 2.73 0 4.1-0.03-53.74-38.84-42.04-193.68-85.44-204.35z"
        fill="url(#linearGradient2669)"
        fillRule="evenodd"
        id="path3412"
      />
      <path
        d="m60.321 136.15c-19.536 0.62-42.335 10.86-53.063 33.13l-0.5622 1.34c-0.3594 4.62-0.5625 9.29-0.5625 14 0 44.15 16.068 84.55 42.657 115.72-24.609-77.25 50.266-145.17 42.125-152.72-5.382-7.8-17.227-11.89-30.594-11.47z"
        fill="url(#linearGradient2671)"
        fillRule="evenodd"
        id="path3422"
      />
    </g>
  </g>
</svg>
`;

exports[`DisplayBall renders correctly empty circle 1`] = `
<svg>
  <circle
    cx="12"
    cy="12"
    fill="white"
    r="8"
    stroke="white"
    strokeWidth="6"
  />
</svg>
`;

exports[`DisplayBall renders correctly isPlayerTwoBall 1`] = `
<svg
  className="balls"
  version="1.0"
  viewBox="0 0 1280 1276"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="#000000"
    stroke="none"
    transform="translate(0.000000,1276.000000) scale(0.100000,-0.100000)"
  >
    <path
      d="M6110 12749 c-36 -4 -153 -15 -260 -24 -794 -63 -1576 -283 -2307
 -648 -265 -133 -631 -347 -593 -347 8 0 76 7 150 15 309 34 769 29 1083 -11
 671 -85 1276 -302 1900 -681 78 -47 275 -175 438 -284 441 -294 542 -345 659
 -337 74 5 181 72 303 188 108 102 187 205 249 326 89 171 127 321 135 524 10
 272 -56 519 -214 796 -47 83 -194 285 -270 372 l-39 44 -139 18 c-77 10 -210
 23 -295 29 -85 6 -191 16 -236 21 -93 12 -469 11 -564 -1z"
    />
    <path
      d="M7650 12624 c0 -4 26 -48 58 -98 261 -407 385 -767 383 -1108 -1
 -422 -181 -790 -601 -1232 -61 -64 -108 -121 -105 -126 5 -8 857 -806 931
 -872 21 -18 23 -17 275 184 836 664 1162 907 1544 1145 210 132 353 212 558
 314 92 46 167 89 167 95 0 23 -267 264 -490 443 -726 580 -1654 1026 -2540
 1221 -80 17 -153 34 -162 36 -10 3 -18 2 -18 -2z"
    />
    <path
      d="M3090 11609 c-143 -14 -240 -29 -348 -55 -95 -22 -109 -29 -185 -86
 -402 -303 -813 -700 -1124 -1084 -658 -814 -1100 -1756 -1302 -2774 -32 -158
 -71 -405 -71 -444 0 -11 25 30 55 92 239 494 762 872 1650 1191 554 200 1076
 332 2392 607 1094 228 1404 303 1783 433 285 97 467 185 610 292 99 75 159
 143 202 228 27 54 32 78 33 132 0 63 -2 70 -48 137 -279 408 -1256 924 -2237
 1181 -496 130 -1034 187 -1410 150z"
    />
    <path
      d="M10885 10627 c-760 -488 -1605 -1109 -2172 -1596 -68 -57 -123 -108
 -123 -113 0 -7 775 -986 811 -1022 18 -19 21 -18 178 57 969 466 1729 499
 2365 105 250 -155 508 -417 709 -720 40 -59 73 -105 75 -103 6 6 -26 211 -59
 376 -213 1073 -707 2086 -1418 2908 -168 193 -185 211 -196 211 -5 0 -82 -47
 -170 -103z"
    />
    <path
      d="M7090 9793 c-348 -295 -883 -552 -1562 -748 -410 -119 -693 -184
 -1598 -366 -789 -159 -998 -204 -1350 -295 -952 -244 -1567 -531 -2032 -948
 -247 -221 -511 -590 -531 -741 -15 -117 -20 -429 -8 -575 7 -80 16 -201 22
 -270 98 -1273 587 -2495 1402 -3503 64 -79 81 -96 93 -86 7 6 100 117 206 248
 1396 1717 2614 3036 4208 4557 461 440 1220 1122 1839 1653 l283 244 -443 443
 c-244 244 -448 444 -454 443 -5 0 -39 -25 -75 -56z"
    />
    <path
      d="M8125 8508 c-1743 -1318 -3436 -2971 -5486 -5353 -382 -445 -939
 -1106 -939 -1116 0 -4 73 -80 163 -170 988 -995 2282 -1645 3642 -1829 83 -11
 173 -25 202 -31 51 -12 51 -12 155 41 611 313 1019 920 1338 1996 129 431 204
 761 420 1829 112 559 139 687 190 915 321 1448 707 2289 1272 2775 64 55 117
 104 118 108 2 12 -813 1016 -825 1017 -5 0 -118 -82 -250 -182z"
    />
    <path
      d="M10570 8083 c-381 -57 -619 -168 -775 -360 -46 -56 -47 -59 -42 -114
 9 -102 58 -191 407 -739 346 -544 494 -797 670 -1148 388 -774 578 -1434 622
 -2153 22 -376 -23 -811 -124 -1177 -17 -62 -29 -115 -27 -117 3 -2 43 44 90
 103 745 929 1217 2054 1354 3226 49 422 63 822 38 1124 -14 160 -20 179 -110
 330 -330 555 -904 919 -1599 1017 -98 14 -432 19 -504 8z"
    />
    <path
      d="M9232 7296 c-273 -108 -514 -417 -705 -901 -173 -440 -301 -929 -513
 -1950 -225 -1084 -292 -1388 -391 -1780 -263 -1043 -561 -1756 -931 -2235 -79
 -101 -274 -304 -349 -362 -39 -30 -50 -44 -41 -50 15 -9 321 -2 543 13 1182
 81 2313 488 3275 1178 254 182 449 344 703 584 137 129 139 131 192 241 318
 662 282 1650 -105 2807 -315 946 -852 1912 -1303 2346 -115 110 -134 120 -247
 129 -53 3 -76 0 -128 -20z"
    />
  </g>
</svg>
`;
